### ISSUE TITLE
Layers, loss functions, datasets, and models for Single Image Super-Resolution (SISR)

### ISSUE THREAD

**Issue by mihirparadkar** [keras-team/keras-cv#448]
**Short Description**
Single-Image Super-Resolution describes the domain of enhancing image resolution for single images (as opposed to groups of images of a scene, for example). Solutions in this domain have applications in security, medical imaging, segmentation, and video enhancement.

**Papers**
Survey paper (2021): [From Beginner to Master: A Survey for Deep
Learning-based Single-Image Super-Resolution](https://arxiv.org/pdf/2109.14335.pdf)

*Datasets:*
[Berkeley Segmentation Data Set and Benchmarks 500 (BSDS500)](https://www2.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/resources.html#bsds500)

[NTIRE 2017 Challenge on Single Image Super-Resolution: Dataset and Study](https://people.ee.ethz.ch/~timofter/publications/Agustsson-CVPRW-2017.pdf)
[Div2K Dataset](https://data.vision.ee.ethz.ch/cvl/DIV2K/)

*Layers*
Sub-pixel convolution: [Real-Time Single Image and Video Super-Resolution Using an Efficient
Sub-Pixel Convolutional Neural Network](https://arxiv.org/pdf/1609.05158.pdf)

*Loss Functions*
Gradient Prior Loss
Edge Prior Loss

*Metrics*
Peak Signal-to-Noise Ratio (PSNR)
Structural Similarity index measure (SSIM)

**Existing Implementations**
https://keras.io/examples/vision/super_resolution_sub_pixel/
https://github.com/krasserm/super-resolution

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
